### PR TITLE
feat(gitlab): add gitlab ci support

### PR DESCRIPTION
**Pull request by fargito**
Related to: #22

### CODE REVIEW

File: .gitlab-ci.yml
Comment (fargito) [inline .gitlab-ci.yml:1]: This commit needs to be dropped, but I need it for now for testing purposes

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+  "rust-analyzer.check.command": "clippy",
+  "[dotenv][ignore][properties][shellscript]": {
+    "editor.defaultFormatter": "foxundermoon.shell-format"
+  },
+  "[html][javascript][json]": {
+    "editor.defaultFormatter": "esbenp.prettier-vscode"
+  },
+  "[rust]": {
+    "editor.defaultFormatter": "rust-lang.rust-analyzer",
+    "editor.inlayHints.enabled": "offUnlessPressed"
```
Comment (adriencaccia) [inline .vscode/settings.json]: This setting is personal, we can remove it

File: src/run/ci_provider/github_actions/provider.rs [hunk 2]
```diff
@@ -95,16 +96,12 @@ impl TryFrom<&Config> for GitHubActionsProvider {
...
-                    login: get_env_variable("GITHUB_ACTOR")?,
-                    id: get_env_variable("GITHUB_ACTOR_ID")?
-                        .parse()
-                        .context("Failed to parse GITHUB_ACTOR_ID into an integer")?,
-                }),
+                run_id: get_env_variable("GITHUB_RUN_ID")?,
             },
+            sender: Some(Sender {
+                login: get_env_variable("GITHUB_ACTOR")?,
+                id: get_env_variable("GITHUB_ACTOR_ID")?,
```
Comment (adriencaccia) [inline src/run/ci_provider/github_actions/provider.rs:103]: 👍

File: src/run/ci_provider/interfaces.rs [hunk 3]
```diff
@@ -1,16 +1,25 @@
 use serde::{Deserialize, Serialize};
 
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum RepositoryProvider {
+    Gitlab,
+    Github,
```
Comment (adriencaccia) [inline src/run/ci_provider/interfaces.rs]:
```suggestion
    GitLab,
    GitHub,
```

File: src/run/ci_provider/gitlab_ci/logger.rs [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+use crate::{
+    logger::{get_group_event, GroupEvent},
+    run::ci_provider::logger::should_provider_logger_handle_record,
+};
+
+lazy_static! {
+    static ref GITLAB_SECTION_ID_REGEX: Regex =
```
Comment (adriencaccia) [inline src/run/ci_provider/gitlab_ci/logger.rs]: The name of this regex does not describe its purpose, maybe something like `GITLAB_SECTION_ID_SANITIZE_REGEX`

File: src/run/ci_provider/gitlab_ci/logger.rs [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+            }
+            Level::Info => {
+                println!("{message}");
+            }
+            Level::Debug => {
+                println!("{U_ESC}[36m{message}{COLOR_RESET}");
+            }
+            Level::Trace => {
+                println!("{U_ESC}[35m{message}{COLOR_RESET}");
+            }
```
Comment (adriencaccia) [inline src/run/ci_provider/gitlab_ci/logger.rs]: It would be better to have the color codes in variables, instead of having to look at the linked gist

**Review by adriencaccia — CHANGES_REQUESTED** [review]
👏

**Review by adriencaccia — APPROVED** [review]
LGTM, waiting on @art049

File: src/run/check_system.rs [hunk 6]
```diff
@@ -104,6 +104,7 @@ lazy_static! {
             ("Ubuntu", "22.04", "aarch64"),
             ("Debian", "11", "x86_64"),
             ("Debian", "12", "x86_64"),
+            ("Debian GNU/Linux", "12", "x86_64"),
```
Comment (art049) [inline src/run/check_system.rs]: What is different here? Why do we need to define a new entry?

File: src/run/ci_provider/gitlab_ci/logger.rs [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+/// CLI magenta
+///
+/// https://gist.github.com/fnky/458719343aabd01cfb17a3a4f7296797#colors--graphics-mode
+const CLI_MAGENTA: &str = "\x1B[35m";
+
+/// Erase cursor till end of line
+///
+/// https://gist.github.com/fnky/458719343aabd01cfb17a3a4f7296797#erase-functions
+const ERASE_CURSOR: &str = "\x1B[0K";
+
```
Comment (art049) [inline src/run/ci_provider/gitlab_ci/logger.rs]: We should use a library instead, like: https://github.com/CodSpeedHQ/runner/blob/1d9a505e87631e5e71eefb8f80e53ca95e2d8283/src/auth.rs#L45-L51

File: .vscode/settings.json [hunk 8]
```diff
@@ -1,5 +1,21 @@
...
+  },
+  "[rust]": {
+    "editor.defaultFormatter": "rust-lang.rust-analyzer"
+  },
+  "files.watcherExclude": {
+    "**/target/**": true
+  },
+  "cSpell.words": [
+    "codspeed"
+  ]
```
Comment (art049) [inline .vscode/settings.json]: can we put that into a separate json file as we have in our monorepo setup?

**Review by art049 — COMMENTED** [review]
WIP review

File: src/run/check_system.rs (hunk 6 above)
Comment (fargito) [reply src/run/check_system.rs]: This is the way it is set by GitlabCI

Comment (fargito) [reply src/run/check_system.rs]: Do you think I should reformat it?

Comment (art049) [reply src/run/check_system.rs]: as discussed it would be great to use `distribution_id` from SystemInfo

File: src/run/runner/valgrind/setup.rs [hunk 9]
```diff
@@ -44,9 +44,10 @@ fn get_codspeed_valgrind_filename(system_info: &SystemInfo) -> Result<String> {
         system_info.os_version.as_str(),
         system_info.arch.as_str(),
     ) {
-        ("Ubuntu", "20.04", "x86_64") | ("Debian", "11", "x86_64") | ("Debian", "12", "x86_64") => {
-            ("20.04", "amd64")
-        }
+        ("Ubuntu", "20.04", "x86_64")
+        | ("Debian", "11", "x86_64")
+        | ("Debian", "12", "x86_64")
+        | ("Debian GNU/Linux", "12", "x86_64") => ("20.04", "amd64"),
```
Comment (art049) [inline src/run/runner/valgrind/setup.rs]: maybe here as well we can use distribution id

**Review by art049 — CHANGES_REQUESTED** [review]
Looking forward to see how the color stuff we discussed will end up being

**Review by art049 — APPROVED** [review]

**Pull request merged**